ui/src/components/admin/ModelsManager/ModelColumns.js

```diff
@@ -42,19 +43,50 @@ const selectedColumns = [
   'updatedBy',
 ];
 
-export const modelStatusPill = (data = null) => {
-  switch (data.status) {
+export const ModelStatusPill = ({ data = null }) => {
+  switch (data) {
     case modelStatus.published:
       return <SmallPill>{modelStatus.published}</SmallPill>;
     case modelStatus.unpublishedChanges:
       return <SmallPill warning>{modelStatus.unpublishedChanges}</SmallPill>;
     case modelStatus.unpublished:
       return <SmallPill info>{modelStatus.unpublished}</SmallPill>;
     default:
-      return <SmallPill primary>{data.status}</SmallPill>;
+      return <SmallPill primary>{data}</SmallPill>;
   }
 };
 
+export const ModelStatusPillWithPublish = ({ name, data }) => {
```

Comment:
Enhanced version of the `ModelStatusPill` component which monitors publish progress.

Since we didn't want the added work of modifying the CMS and adding a new `publishing` state, but we still want the user to know if a model is currently being published, we can track if the model is being published and temporarily update the status here to `Publish in Progress`